### PR TITLE
NEW ERT: Memetic Response Team, and now Memorizer flash works!

### PR DESCRIPTION
**Pull request by DrDiasyl**
## About The Pull Request
Adds new ERT for admins to send in particular situations when the crew needs to forget what happened, either for the well-being of themselves or the corporate.

### The MEMETIC RESPONSE TEAM
memetic agents
_(Blue alert ones on left, Red alerts ones on the right)_

### They are equipped with the next:
**AGENT:**

- Handcuffs
- Mandatory black tie and sunglasses
- Advanced Health Analyzer
- Decloner
- Telescopic Baton
- Memorizer Flash! _(About it later)_
- SWAT Helmet (on Red)
- Reactive Teleport Armor (on Red)
- Cryo Temperature Gun (on Red)

**LEADER**

- Same as the agent
- RED tie and BIGGER sunglasses
- Hypnotic Flashbang
- Reactive Repulse Armor (on Red)
- Shrink Ray (on Red)

### The Memorizer flash
forget-it-flash

Memorizer flash was an item that was in-game for a long time, but did absolutely nothing! It worked just as a normal flash.
Now Memorizer flash "erases" memories, by alt-clicking it you can set the time period you want the victim to forget (Seconds, Minutes, Hours, Days; from 1 to 60). After you flash the victim a large message will appear for them:

**"This light makes you forget everything you saw in the past 2 hours..."** (a time chosen for example)

Memorizer flash is also a sub-type of hypnoflash (but without immense amounts of various status effects, and without requirements for the victim to be sleeping, or drugged), so right after you erased their memories, you can put new ones in their heads.
As this is now pretty powerful, you no longer can find it just laying in maintenance tunnels.

### Some other stuff
Now Cryo Temperature Gun fires cryo rays instead of freeze rays. Cryo rays unlike freeze rays are much more powerful, and make floors icy for a short period of time.
Cryo Temperature Gun was previously used for the Dr.Freeze Syndie Kit, but syndiekits are dead so who cares? This also gives it more use, than just a lame Temperature Gun which only has a freeze option.

(If you're wondering why I gave them such weird weapons, I gave them the kind of weapons Man in Black would have as this is an obvious reference to them)
## Why It's Good For The Game
Gives admins another ERT to fool around with, especially when they have a flash that "erases" memories of people.
"CentCom sends in such ERT to erase memories of command seeing Captain doing embarrassing things for the corporate" for example.

Gives Memorizer Flash finally a use! Before that, it was a rare maintenance loot that did ABSOLUTELY NOTHING! It was just a fancy-looking flash, even though the description said that it would erase memories.

Gives use for the Cryo Temperature Gun as it was just a dead object after the removal of Syndie Kits.
## Changelog
:cl:
add: Memorizer Flash is now actually "erasing" memories! It also functions as easier to use hypnoflash. Before that, it was just a fancier-looking flash.
del: Memorizer Flash is no longer a maintenance loot.
balance: Now Cryo Temperature Gun shoots cryo rays instead of freeze rays.
admin: New Memetic ERT! Armed with various fancy high-tech gear, and memory erasing flashes. Sent in cases when corporate wants you to forget something.
/:cl:

### CODE REVIEW

**Comment by Helg2** [comment]
> syndiekits are dead

But they aren't?

**Comment by DrDiasyl** [comment]
> > syndiekits are dead
> 
> But they aren't?

If they aren't then eh, we just rebalanced their Cryo Temperature Gun, not a big deal.

**Comment by Cheshify** [comment]
As someone that loves sending ERTs, these will be very fun to use for all parties. Maybe the admin team should be alerted of the possible policy change that comes with this, since that memory thing will need to be enforced.

**Comment by TheVekter** [comment]
This sounds like a cool idea but I've got issues:

1) We already have a lot of ERTs as it is. I'm not necessarily against the idea of having more, and I don't think there's necessarily a negative to it, but feature bloat is a thing that exists. Are you prepared to maintain this further if need be in the future?
2) We already have issues with players respecting the rule about not remembering who sent you to the funny meme hand dimension as Heretic. I don't see this being as big of an issue with how this doesn't directly involve an antag, but it bugs me that we're being given something else kind of nebulous to enforce. How does this otherwise differ from those issues we're having with Heretic?

Otherwise I love the idea and do want to see it added.

**Comment by Striders13** [comment]
setting time that you want the victim to forget, while cool, sounds too complicated for both player (I DONT FUCKING KNOW IF I SAW JOHN TRAITOR BEHEAD THAT GUY 30 MINUTES AGO! WAS IT 40?) and admin to enforce. Just make them forget the entire round.

**Comment by TheVekter** [comment]
> setting time that you want the victim to forget, while cool, sounds too complicated for both player (I DONT FUCKING KNOW IF I SAW JOHN TRAITOR BEHEAD THAT GUY 30 MINUTES AGO! WAS IT 40?) and admin to enforce. Just make them forget the entire round.

Could just mandate that, in order for them to actually forget something, whoever flashes them has to tell them what to forget.

**Comment by DrDiasyl** [comment]
> This sounds like a cool idea but I've got issues:
> 
> 1) We already have a lot of ERTs as it is. I'm not necessarily against the idea of having more, and I don't think there's necessarily a negative to it, but feature bloat is a thing that exists. Are you prepared to maintain this further if need be in the future?
> 2) We already have issues with players respecting the rule about not remembering who sent you to the funny meme hand dimension as Heretic. I don't see this being as big of an issue with how this doesn't directly involve an antag, but it bugs me that we're being given something else kind of nebulous to enforce. How does this otherwise differ from those issues we're having with Heretic?
> 
> Otherwise I love the idea and do want to see it added.

1) Yeah, we really do have plenty of ERTs. Some are used often (Interns) some not (Bounty Hunters), but this one is pretty unique so it can pass.

2) This is ERT, and as ERT it mostly likely won't appear that often. We can handle them arriving on station once 2 weeks, erasing memories of HoP getting captain's ass faxed to them.

**Comment by DrDiasyl** [comment]
> setting time that you want the victim to forget, while cool, sounds too complicated for both player (I DONT FUCKING KNOW IF I SAW JOHN TRAITOR BEHEAD THAT GUY 30 MINUTES AGO! WAS IT 40?) and admin to enforce. Just make them forget the entire round.

Well, if the time thingy deemed too complicated, I can change it to a message one. So you just input in what specifically you want them to forget.

**Comment by TheVekter** [comment]
>     1. Yeah, we really do have plenty of ERTs. Some are used often (Interns) some not (Bounty Hunters), but this one is pretty unique so it can pass.
> 
>     2. This is ERT, and as ERT it mostly likely won't appear that often. We can handle them arriving on station once 2 weeks, erasing memories of HoP getting captain's ass faxed to them.

Cool, I'm sold. Thanks for the response.

**Comment by Cheshify** [comment]
> > setting time that you want the victim to forget, while cool, sounds too complicated for both player (I DONT FUCKING KNOW IF I SAW JOHN TRAITOR BEHEAD THAT GUY 30 MINUTES AGO! WAS IT 40?) and admin to enforce. Just make them forget the entire round.
> 
> Well, if the time thingy deemed too complicated, I can change it to a message one. So you just input in what specifically you want them to forget.
> 
> And also it works as a hypnoflash too, so you anyway kinda tell what memory you're replacing with what memory.

Maybe like, two boxes you can fill in? You forget X and remember Y.

**Comment by DrDiasyl** [comment]
> > > setting time that you want the victim to forget, while cool, sounds too complicated for both player (I DONT FUCKING KNOW IF I SAW JOHN TRAITOR BEHEAD THAT GUY 30 MINUTES AGO! WAS IT 40?) and admin to enforce. Just make them forget the entire round.
> > 
> > Well, if the time thingy deemed too complicated, I can change it to a message one. So you just input in what specifically you want them to forget.
> > 
> > And also it works as a hypnoflash too, so you anyway kinda tell what memory you're replacing with what memory.
> 
> Maybe like, two boxes you can fill in? You forget X and remember Y.

Sounds like a brilliant idea! But this one works just fine for now.

**Comment by DrDiasyl** [comment]
BUT! Having this use the time like it does right now, allows you to do this masterpiece!

_Sets to 5 seconds_
"Hey, look over there!"
_Flash_
"Hey, look over there!"
_Flash_
"Hey, look over there!"
_Flash_
…

**Comment by nukularpower** [comment]
love the idea and also love anything that encourages more ERTs to be sent

**Comment by Elli-Skala** [comment]
If it's a sub type of the hypnotic flash maybe it would be a good idea to give them a radio jammer so they will only hear your "memory"/can't go yelling what's happening to others (they are after all a rumor, recognizable only as deja vu and dismissed just as quickly)

**Comment by carlarctg** [comment]
They should get reactive stealth armor, it's more men in black-ey than repulse or teleport

There's nothing stopping you from keeping the flash memorizer as maint loot that's just for flavor - it'll certainly still be more seen and used than this ERT. Just make it a non hypno type and make the description say it's rusted or whatever.

File: code/modules/projectiles/ammunition/energy/special.dm [hunk 1]
```diff
@@ -42,6 +42,12 @@
 	projectile_type = /obj/projectile/temp/hot
 	select_name = "bake"
 
+/obj/item/ammo_casing/energy/temp/cryo
+	projectile_type = /obj/projectile/temp/cryo
+	select_name = "freeze"
+	e_cost = 250
+	fire_sound = 'sound/weapons/pulse3.ogg'
```
Comment (necromanceranne) [inline code/modules/projectiles/ammunition/energy/special.dm:49]:
```suggestion
```

This is already defined by the parent type.

File: code/datums/ert.dm [hunk 2]
```diff
@@ -116,3 +116,21 @@
...
 	random_names = FALSE
 	ert_template = /datum/map_template/shuttle/ert/bounty
+
+/datum/ert/memetic
+	leader_role = /datum/antagonist/ert/memetic/commander
+	roles = list(/datum/antagonist/ert/memetic)
+	teamsize = 3
+	code = "Blue"
+	rename_team = "Memetic Response Team"
+	mission = "Replace the memories of crew about particular incident."
```
Comment (necromanceranne) [inline code/datums/ert.dm:126]:
```suggestion
	mission = "Censor all information and knowledge related to a highly classified incident aboard the station."
```

**Comment by nukularpower** [comment]
reactive stealth armor does seem more fitting

**Comment by ATH1909** [comment]
> Memorizer flash is also a sub-type of hypnoflash (but without immense amounts of various status effects, and without requirements for the victim to be sleeping, or drugged), so right after you erased their memories, you can put new ones in their heads.

is there an option to NOT put them into a trance after wiping their memories? or, like, a subtype that ONLY wipes memories? needing to say a dummy message for a wipee to latch onto might get annoying, and if an antag nabs one of these, they're basically gonna turn into a headrev.

**Comment by dessysalta** [comment]
image

**Comment by HoodMilk** [comment]
Please we need a necromancer, this deserves to live.

**Comment by dessysalta** [comment]
image

**Pull request closed without merging**